### PR TITLE
feat(id-validations):

### PR DESCRIPTION
**Pull request by daniel7764**
Can't save contacted people with the same identification number/identification number that already exists in the investigation

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts [hunk 1]
```diff
@@ -42,14 +62,18 @@ const useContactQuestioning = (parameters: useContactQuestioningParameters): use
...
         }).catch(err => {
-            logger.error({
-                service: Service.CLIENT,
-                severity: Severity.LOW,
-                workflow: 'Saving single contact',
-                step: `got the following error from the server: ${err}`,
-                user: userId,
-                investigation: epidemiologyNumber
-            })
+            if (err.response.data === 'found duplicate ids') {
```
Comment (roymaniv) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts:65]: const

File: server/src/DBService/ContactedPeople/Mutation.ts [hunk 2]
```diff
@@ -7,3 +7,11 @@ mutation updateAllUnSavedContacts($unSavedContacts: JSON!) {
   }
 }
 `;
+
+export const CHECK_FOR_DUPLICATE_IDS = gql`
```
Comment (OrelKaravani) [inline server/src/DBService/ContactedPeople/Mutation.ts:11]: I would call it "CHECK_FOR_DUPLICATE_CONTACTED_PERSON_IDS" to be clear that it checks contacted person ids and not just random ids.

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts [hunk 3]
```diff
@@ -1,24 +1,44 @@
...
+    const handleDuplicateIdsError = () => {
+        logger.info({
+            service: Service.CLIENT,
+            severity: Severity.LOW,
+            workflow: 'Create Interaction',
+            step: 'Didnt save interactions due to duplicate ids',
+            user: userId,
+            investigation: epidemiologyNumber
+        });
+        Swal.fire({
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts:32]: Use custom swal

**Review by eladyashar — DISMISSED** [review]

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/useInteractionsForm.ts [hunk 4]
```diff
@@ -1,132 +1,181 @@
...
+            if (interactionsDataToSave[InteractionEventDialogFields.ID]) {
+                logger.info({
+                    service: Service.CLIENT,
+                    severity: Severity.LOW,
+                    workflow: 'Update Interaction',
+                    step: `launching update interaction request`,
+                    user: userId,
+                    investigation: epidemiologyNumber
+                });
+                axios.post('/intersections/updateContactEvent', {
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/useInteractionsForm.ts:44]: If you parse the data the same way on both update and create, maybe you should do it before the condition

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/useInteractionsForm.ts [hunk 5]
```diff
@@ -1,132 +1,181 @@
...
+                    .catch((error) => {
+                        if (error.response.data === 'found duplicate ids') {
+                            logger.info({
+                                service: Service.CLIENT,
+                                severity: Severity.LOW,
+                                workflow: 'Create Interaction',
+                                step: 'Didnt save interactions due to duplicate ids',
+                                user: userId,
+                                investigation: epidemiologyNumber
+                            });
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/useInteractionsForm.ts:114]: Use custom swal

Comment (MayaSasson) [reply client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/useInteractionsForm.ts:114]: In every swal

File: client/src/components/App/Content/InvestigationForm/useInvestigationForm.ts [hunk 6]
```diff
@@ -218,7 +218,7 @@ const useInvestigationForm = (): useInvestigationFormOutcome => {
 
     useEffect(() => {
         if (epidemiologyNumber !== defaultEpidemiologyNumber && userId !== defaultUser.id)
-        fetchCities();
+            fetchCities();
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/useInvestigationForm.ts:221]: Remove tab

File: server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts [hunk 7]
```diff
@@ -143,21 +144,32 @@ ContactedPeopleRoute.post('/interactedContacts', (request: Request, response: Re
...
+            } else {
+                logger.info({
+                    service: Service.SERVER,
+                    severity: Severity.LOW,
+                    workflow,
+                    step: 'found duplicate ids in the DB',
+                    user: response.locals.user.id,investigation: epidemiologyNumber
+                });
+                response.status(500).send('found duplicate ids');
+            }
```
Comment (MayaSasson) [inline server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts:172]: Use const for 500

File: server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts [hunk 8]
```diff
@@ -143,21 +144,32 @@ ContactedPeopleRoute.post('/interactedContacts', (request: Request, response: Re
...
-            logger.info({
-                service: Service.SERVER,
-                severity: Severity.LOW,
-                workflow,
-                step: 'got response from DB',
-                user: response.locals.user.id,investigation: epidemiologyNumber
-            });
-            isThereDoneContact && sendSavedInvestigationToIntegration(epidemiologyNumber, workflow, response.locals.user.id);
-            response.send(result);
+            if(!result.errors) {
```
Comment (MayaSasson) [inline server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts:153]: Is this the check you want to commit? I think that checking if the result structure is as expected is more accurate

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/useInteractionsForm.ts (hunk 5 above)
Comment (daniel7764) [reply client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/useInteractionsForm.ts:114]: in this swal im also adding a class for the container so i can't use the custom swal from the commons

File: server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts [hunk 9]
```diff
@@ -143,21 +144,32 @@ ContactedPeopleRoute.post('/interactedContacts', (request: Request, response: Re
...
+                    user: response.locals.user.id,investigation: epidemiologyNumber
+                });
+                isThereDoneContact && sendSavedInvestigationToIntegration(epidemiologyNumber, workflow, response.locals.user.id);
+                response.send(result);
+            } else {
+                logger.info({
+                    service: Service.SERVER,
+                    severity: Severity.LOW,
+                    workflow,
+                    step: 'found duplicate ids in the DB',
```
Comment (MayaSasson) [inline server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts:168]: I don’t think that the request is necessarily failed by duplicate keys, this is an inaccurate assumption.
Change the step to print the error on the result object

File: server/src/ClientToDBAPI/InvestigationInfo/mainRoute.ts [hunk 10]
```diff
@@ -159,44 +160,50 @@ investigationInfo.post('/updateInvestigationStatus', (request: Request, response
...
+                                    severity: Severity.HIGH,
+                                    workflow: 'Ending Investigation',
+                                    step: `failed to update the investigation end time due to: ${err}`,
+                                    user: response.locals.user.id,
+                                    investigation: epidemiologyNumber
+                                });
+                                response.status(errorStatusCode).json({message: 'failed to update the investigation end time'});
+                            });
+                        } else {
+                            response.status(500).send('found duplicate ids');
```
Comment (MayaSasson) [inline server/src/ClientToDBAPI/InvestigationInfo/mainRoute.ts:204]: Send the message on the same way as line 201, so schemas won’t have to change

Comment (MayaSasson) [inline server/src/ClientToDBAPI/InvestigationInfo/mainRoute.ts:204]: Send the message on the same way as line 201, so schemas won’t have to change

File: server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts [hunk 11]
```diff
@@ -103,7 +104,7 @@ ContactedPeopleRoute.get('/amountOfContacts/:investigationId', (request: Request
 ContactedPeopleRoute.get('/allContacts/:investigationId', (request: Request, response: Response) => {
     const getContactsQueryVariables = { investigationId: parseInt(request.params.investigationId) }
     logger.info({
-        service: Service.SERVER,
+        service: Service.CLIENT,
```
Comment (ItayBenMoshe) [inline server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts:107]: why client?

File: server/src/ClientToDBAPI/IntersectionsRoute/mainRoute.ts [hunk 12]
```diff
@@ -185,33 +224,37 @@ intersectionsRoute.post('/updateContactEvent', (request: Request, response: Resp
...
-    .catch(err => {
-        logger.error({
-            service: Service.SERVER,
-            severity: Severity.LOW,
-            workflow: 'Update Contact Event',
-            step: `got errors approaching the graphql API ${err}`,
-            investigation: response.locals.epidemiologynumber,
-            user: response.locals.user.id
+        .then(result => {
+            if(result.data.updateContactEventFunction) {
```
Comment (MayaSasson) [inline server/src/ClientToDBAPI/IntersectionsRoute/mainRoute.ts:233]: change to result?.data?.updateContactEventFunction

File: server/src/ClientToDBAPI/IntersectionsRoute/mainRoute.ts [hunk 13]
```diff
@@ -148,33 +183,37 @@ intersectionsRoute.post('/createContactEvent', (request: Request, response: Resp
...
-            service: Service.SERVER,
-            severity: Severity.LOW,
-            workflow: 'Create Contact Event',
-            step: 'got response from DB',
-            investigation: response.locals.epidemiologynumber,
-            user: response.locals.user.id
-        });
-        response.send(result);
+        .then(result => {
+            if (result.data.updateContactEventFunction) {
```
Comment (MayaSasson) [inline server/src/ClientToDBAPI/IntersectionsRoute/mainRoute.ts:192]: change to result?.data?.updateContactEventFunction

**Review by roymaniv — DISMISSED** [review]

**Review by ItayBenMoshe — APPROVED** [review]

**Review by roymaniv — APPROVED** [review]

**Pull request merged**